src/compute-client/src/controller/instance.rs

```diff
@@ -534,17 +534,31 @@ impl<T: ComputeControllerTimestamp> Instance<T> {
         );
     }
 
-    /// Returns `true` iff all (non-transient) collections are hydrated on at
-    /// least one (possibly different) replica.
+    /// Returns `true` iff each (non-transient) collection is hydrated on at
+    /// least one replica.
     ///
     /// This also returns `true` in case this cluster does not have any
     /// replicas.
-    pub fn all_collections_hydrated(&self) -> bool {
+    pub fn all_collections_hydrated_on_replicas(
+        &self,
+        target_replica_ids: Option<Vec<ReplicaId>>,
+    ) -> bool {
         if self.replicas.is_empty() {
             return true;
         }
-
         let mut all_hydrated = true;
+        let target_replicas: BTreeSet<ReplicaId> = self
+            .replicas
+            .keys()
+            .filter_map(|id| match target_replica_ids {
+                None => Some(id.clone()),
+                Some(ref ids) if ids.contains(id) => Some(id.clone()),
+                Some(_) => None,
+            })
+            .collect();
+        if target_replicas.is_empty() {
```

Comment:
This case only happens when the set of passed-in replicas does not overlap with the replicas of this cluster. Is that maybe user error and we should report an error? Current behavior is also fine for me, though.